common/src/main/scala/dx/util/JsUtils.scala

```diff
@@ -220,18 +220,21 @@ object JsUtils {
   // Make a JSON value deterministically sorted.  This is used to
   // ensure that the checksum does not change when maps
   // are ordered in different ways.
-  //
-  // Note: this does not handle the case of arrays that
-  // may have different equivalent orderings.
+
   def makeDeterministic(jsValue: JsValue): JsValue = {
     jsValue match {
       case JsObject(m: Map[String, JsValue]) =>
         // deterministically sort maps by using a tree-map instead
         // a hash-map
         val mTree = m
-          .map { case (k, v) => k -> JsUtils.makeDeterministic(v) }
+          .map {
+            case (k, JsArray(a)) =>
+              k -> JsArray(a.map(JsUtils.makeDeterministic).sortWith(_.toString < _.toString))
+            case (k, v) => k -> JsUtils.makeDeterministic(v)
```

Comment:
This LGTM.

I'm wondering if we need a recursive check, for cases when an array is embedded deeper? E.g. (k, v), where v is another map (q, r), where r is a map, etc...

Reply:
Added. Thanks for suggesting it :D

Reply:
Great, thank you @YuxinShi0423!